asf_heat_pump_suitability/analysis/exploratory/create_full_dataset/assign_cluster_suitability_and_feasibility.py

```diff
@@ -0,0 +1,237 @@
+            )
+        )
+        .alias("suitable_scheme")
+    )
+
+    return cluster_df
+
+
+def create_df_feasibility_scoring(
+    df: pl.DataFrame,
+    weights: dict,
+    cols_to_aggregate: list = None,
+    anchor_loads_threshold: float = 500,
+    city_centre_threshold: float = 500,
+) -> pl.DataFrame:
+    """
+    Computes the feasibility score for each scheme and cluster in the dataset.
+
+    Schemes include:
+    - Individual Air Source Heat Pump (ASHP)
+    - Collective purchasing of ASHP
+    - Shared Ground Loop (SGL)
+    - Heat Network (HN)
+
+    Args:
+        df (pl.DataFrame): DataFrame containing the dataset with feature values for each UPRN.
+        weights (dict): Dictionary containing the weights for each scheme and features used in feasibility scoring.
+            'weights' should contain the following keys (schemes):
+                "individual_ashp_feasibility", "collective_ashp_feasibility", "sgl_feasibility" and "hn_feasibility"
+            The value corresponding to each scheme should be a dictionary whith where keys subsets of features in cols_to_aggregate
+        cols_to_aggregate (list, optional): List of columns to aggregate. If None, defaults to a predefined list.
+        anchor_loads_threshold (float, optional): default threhold for distance to anchor loads in meteres. Defaults to 500 m.
+        city_centre_threshold (float, optional): default threhold for distance to city center in meteres. Defaults to 500 m.
+
+    Raises:
+        ValueError: if schemes or features provided in weights do not exist.
+
+    Returns:
+        pl.DataFrame: DataFrame with aggregated feasibility scores for each cluster.
+    """
+
+    if cols_to_aggregate is None:
+        # Creating a column for not in conservation area
+        df = df.with_columns((~pl.col("in_cons_area")).alias("not_in_cons_area"))
+
+        # Creating close_to_anchor_loads from distance_to_anchor_loads and anchor_loads_threhold
+        df = df.with_columns(
+            (pl.col("distance_to_anchor_loads") <= anchor_loads_threshold).alias(
+                "close_to_anchor_loads"
+            )
+        )
+        # Creating close_to_city_centre from distance_to_city_centre and city_centre_threshold
+        df = df.with_columns(
+            (pl.col("distance_to_city_centre") <= city_centre_threshold).alias(
+                "close_to_city_centre"
+            )
+        )
+
+        # Columns to compute percentages for feasibility scoring
+        cols_to_aggregate = [
+            "owner_occupied",
+            "high_income_decile",
+            "on_gas",
+            "not_listed",
+            "not_in_cons_area",
+            "social_housing",
+            "flats",
+            "on_communal_heating",
+            "has_outdoor_space",
+            "in_listed_buildings",
+            "in_cons_area",
+            "in_hn",
+            "close_to_anchor_loads",
+            "close_to_city_centre",
+        ]
```

Comment:
Once @crispy-wonton finishes her work on the plymouth data, I can improve this part.